src/SegmentedString.chpl

```diff
@@ -1112,6 +1117,59 @@ module SegmentedString {
       return ranks;
     }
 
+    proc getFixesHelper(n: int, kind: Fixes, proper: bool) {
+      const lengths = getLengths() - 1;
+      var longEnough = makeDistArray(size, bool);
+      if proper {
+        longEnough = (lengths > n);
+      } else {
+        longEnough = (lengths >= n);
+      }
+      var nFound = + reduce longEnough;
+      var retOffsets = makeDistArray(nFound, int);
+      forall (i, o) in zip(retOffsets.domain, retOffsets) {
+        o = i * (n + 1);
+      }
+      var retBytes = makeDistArray(nFound * (n + 1), uint(8));
+      var srcInds = makeDistArray(nFound * (n + 1), int);
```

Comment:
Are we guaranteed that `retBytes` and `srcInds` will have the same domain if we do it this way? And does that even matter? If that is something to worry about, maybe we do
```chapel
const retDom = makeDistDom(nFound * (n + 1));
var retBytes = [retDom] uint(8);
var srcInds = [retDom] int;
```

Reply:
Your way is safer and more maintainable, so I'll go with it, although I don't think the way it's written carries much risk. Unless something fundamental changes in `makeDistArray`, those two arrays are guaranteed to have equivalent domains (although not the same domain object in memory, I don't think). Having equivalent domains is important, because I don't think you can `zip` different domains together in a `forall`.